### PR TITLE
[MB-14005] shipmentInfoList components refactor

### PR DESCRIPTION
**Pull request by SirenaBorracha**
## [Jira ticket](https://dp3.atlassian.net/browse/MB-14005) for this change

This work was discovered as a result of [MB-13322](https://dp3.atlassian.net/browse/MB-13322). That story is marked 'not doing' and this PR is an attempt to prevent future bugs being reported wrt which fields are visible in the shipment details accordion depending on shipment type. Conversation for which fields should show when collapsed/expanded [here](https://ustcdp3.slack.com/archives/CP4979J0G/p1657146426215509?thread_ts=1657136724.161959&cid=CP4979J0G).

## Summary

This PR refactors the NTS and NTSr shipment info list components, removing superfluous code to render the `mtoAgents`. NTS shipments do not have a `receivingAgent` because that role is filled by the storage company. Inversely, NTSR shipments do not have a `releasingAgent` as that role is again filled by the storage company. Additionally I've updated the storybook titles for the shipment info list components to minimize confusion when looking for a particular shipment type.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use two terminals to test this locally.</summary>

##### Terminal 1

Start the Go server locally.

```sh
make server_run
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

</details>

### Additional steps

1. Access the Office app
2. Login as a TOO/TIO (I used too_tio_role@office.mil)
3. Search for move code `NTSEVR` which has both NTS and NTSR shipments, navigate to the Move Details page
4. Expand and collapse each accordion to see the NTS displays the value of the Releasing agent and the NTSR shows a Receiving agent
5. Navigate back to the Moves Queue, search for move code `REQSRV` which has HHG and NTSR shipments
6. Expand and collapse each accordion to see the placeholder `—` value for the Receiving agent field

An additional check would be to log in as QAE/CSR (qae_csr_role@office.mil) and search the above (or for other moves) to confirm they are correctly rendering fields in shipment details.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Request review from a member of a different team.
- [N/A] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [N/A] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### Screenshots

Move Code `NTSEVR`
shipmentInfoLists-refactor

Move Code `REQSRV`
shipmentInfoLists-refactor

### CODE REVIEW

File: src/components/Office/DefinitionLists/NTSRShipmentInfoList.jsx [hunk 1]
```diff
@@ -53,6 +52,10 @@ const NTSRShipmentInfoList = ({
   });
   setDisplayFlags(errorIfMissing, warnIfMissing, showWhenCollapsed, null, shipment);
 
+  const showElement = (elementFlags) => {
+    return (isExpanded || elementFlags.alwaysShow) && !elementFlags.hideRow;
+  };
+
```
Comment (PDickeyTruss) [inline src/components/Office/DefinitionLists/NTSRShipmentInfoList.jsx:58]: Nice addition - this really cleaned up all the inline conditionals below 🎉

**Review by PDickeyTruss — APPROVED** [review]
LGTM! Code changes look good. Works/displays as described in the PR. Approved ✅

**Review by YanZ777 — APPROVED** [review]
Nice cleanup! Code looks good to me!

**Review by cosburn — APPROVED** [review]
Looks good!

**Pull request merged**